app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/UserPointTest.java

```diff
@@ -0,0 +1,87 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.rule.ServiceTestRule;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Random;
+import java.util.concurrent.TimeoutException;
+
+
+public class UserPointTest {
+
+    @Rule
+    public final ServiceTestRule serviceRule = new ServiceTestRule();
+
+    @Test
+    public void getLocationTest() throws TimeoutException {
+
+        serviceRule.startService(
+                new Intent(ApplicationProvider.getApplicationContext(),
+                        UserPoint.class));
+
+        Context mContext = ApplicationProvider.getApplicationContext();
+
+        UserPoint userPoint = new UserPoint(mContext);
+
+        if (userPoint.canGetLocation()) {
+            return;
+        } else {
+            Assert.assertEquals(27.988056, userPoint.getLatitude(), 0);
+            Assert.assertEquals(86.925278, userPoint.getLongitude(), 0);
+            Assert.assertEquals(8848.86, userPoint.getAltitude(), 0);
+            Assert.assertEquals(0.0, userPoint.getAccuracy(), 0);
+        }
+
+    }
+
+    @Test
+    public void setLocationTest() throws TimeoutException {
+
+        serviceRule.startService(
+                new Intent(ApplicationProvider.getApplicationContext(),
+                        UserPoint.class));
+
+        Context mContext = ApplicationProvider.getApplicationContext();
+
+        UserPoint userPoint = new UserPoint(mContext);
+
+        Random r = new Random();
+
+        double lat = r.nextDouble();
+        double lon = r.nextDouble();
+        double alt = r.nextDouble();
+        double acc = r.nextDouble();
```

Comment:
I see the point with trying random values all the time. But as the tests are already slow, maybe just setting a constant is better for performance. It shouldn't differ the test if we always try a constant compared to a value between 0 and 1.

Reply:
If I understood correctly, this type of random values don't impact the performance at all